broker-daemon/utils/cached-call.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Default time for the cached call to live. Calls made after this time will
+ * make another call to the promise function.
+ * @constant
+ * @type {number}
+ * @default
+ */
+const TIME_TO_LIVE = 5000
+
+/**
+ * @class Create a wrapper for a call to a function that returns a promise.
+ * Subsequent calls to the function will return the cached result if the
+ * previous call is within the time to live.
+ */
+class CachedCall {
+  /**
+   * Create a new CachedCall instance.
+   * @param {Function} promiseFn - function that returns a promise
+   * @param {number} ttl - milliseconds for previous calls to remain valid
+   */
+  constructor (promiseFn, ttl = TIME_TO_LIVE) {
+    this.promiseFn = promiseFn
+    this.lastCallTime = 0
+    this.lastCallPromise = undefined
```

Comment:
Setting this to null should have the same effect instead of defining a value as undefined XD